File: backend/src/main/java/com/musicflow/service/PlaylistService.java
package com.musicflow.service;

import com.musicflow.entity.Playlist;
import com.musicflow.entity.Song;
import com.musicflow.entity.User;
import com.musicflow.repository.PlaylistRepository;
import com.musicflow.repository.SongRepository;
import com.musicflow.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

@Service
@RequiredArgsConstructor
public class PlaylistService {
    
    private final PlaylistRepository playlistRepository;
    private final UserRepository userRepository;
    private final SongRepository songRepository;
    
    public List<Playlist> getUserPlaylists(Long userId) {
        User user = userRepository.findById(userId)
                .orElseThrow(() -> new RuntimeException("用户不存在"));
        return playlistRepository.findByUser(user);
    }
    
    public Playlist createPlaylist(Long userId, String name, String description) {
        User user = userRepository.findById(userId)
                .orElseThrow(() -> new RuntimeException("用户不存在"));
        
        Playlist playlist = new Playlist();
        playlist.setName(name);
        playlist.setDescription(description);
        playlist.setUser(user);
        
        return playlistRepository.save(playlist);
    }
    
    public Playlist addSongToPlaylist(Long playlistId, Long songId) {
        Playlist playlist = playlistRepository.findById(playlistId)
                .orElseThrow(() -> new RuntimeException("歌单不存在"));
        
        Song song = songRepository.findById(songId)
                .orElseThrow(() -> new RuntimeException("歌曲不存在"));
        
        if (!playlist.getSongs().contains(song)) {
            playlist.getSongs().add(song);
            playlist.setSongCount(playlist.getSongCount() + 1);
            return playlistRepository.save(playlist);
        }
        
        return playlist;
    }
    
    public void deletePlaylist(Long playlistId) {
        playlistRepository.deleteById(playlistId);
    }
    
    public List<Playlist> searchPlaylists(String keyword) {
        return playlistRepository.findByNameContainingIgnoreCase(keyword);
    }
    
    public Page<Playlist> findAllPublic(Pageable pageable) {
        return playlistRepository.findByIsPublicTrue(pageable);
    }
}

File: src/services/playlistService.ts
import api from '@/utils/api'

export interface Playlist {
  id: number
  name: string
  description: string
  coverUrl: string
  songCount: number
  isPublic: boolean
  createdAt: string
}

export interface PlaylistRequest {
  name: string
  description?: string
  isPublic?: boolean
}

export const playlistService = {
  // 获取用户歌单
  async getUserPlaylists(): Promise<Playlist[]> {
    const response = await api.get('/playlists/user')
    return response.data
  },

  // 创建歌单
  async createPlaylist(data: PlaylistRequest): Promise<Playlist> {
    const response = await api.post('/playlists', data)
    return response.data
  },

  // 更新歌单
  async updatePlaylist(id: number, data: PlaylistRequest): Promise<Playlist> {
    const response = await api.put(`/playlists/${id}`, data)
    return response.data
  },

  // 删除歌单
  async deletePlaylist(id: number): Promise<void> {
    await api.delete(`/playlists/${id}`)
  },

  // 获取歌单详情
  async getPlaylistDetail(id: number): Promise<Playlist> {
    const response = await api.get(`/playlists/${id}`)
    return response.data
  },

  // 添加歌曲到歌单
  async addSongToPlaylist(playlistId: number, songId: number): Promise<void> {
    await api.post(`/playlists/${playlistId}/songs/${songId}`)
  },

  // 从歌单移除歌曲
  async removeSongFromPlaylist(playlistId: number, songId: number): Promise<void> {
    await api.delete(`/playlists/${playlistId}/songs/${songId}`)
  }
}